unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/Services/SceneControllerService/SceneControllerServiceImpl.cs

```diff
@@ -0,0 +1,264 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Controllers;
+using DCL.CRDT;
+using DCL.Helpers;
+using DCL.Models;
+using Decentraland.Renderer.RendererServices;
+using Google.Protobuf;
+using KernelCommunication;
+using rpc_csharp;
+using RPC.Context;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using UnityEngine;
+using BinaryWriter = KernelCommunication.BinaryWriter;
+
+namespace RPC.Services
+{
+    public class SceneControllerServiceImpl : IRpcSceneControllerService<RPCContext>
+    {
+        // HACK: Until we fix the code generator, we must replace all 'Decentraland.Common.Entity' for 'DCL.ECSComponents.Entity' in RpcSceneController.gen.cs
+        // to be able to access request.Entity properties.
+        private static readonly UnloadSceneResult defaultUnloadSceneResult = new UnloadSceneResult();
+
+        private const string REQUIRED_PORT_ID_START = "scene-";
+
+        private int sceneNumber = -1;
+        private RPCContext context;
+        private RpcServerPort<RPCContext> port;
+
+        private readonly MemoryStream sendCrdtMemoryStream;
+        private readonly BinaryWriter sendCrdtBinaryWriter;
+        private readonly MemoryStream getStateMemoryStream;
+        private readonly BinaryWriter getStateBinaryWriter;
+
+        private readonly CRDTSceneMessage reusableCrdtMessageResult = new CRDTSceneMessage();
+        private readonly CRDTSceneCurrentState reusableCurrentStateResult = new CRDTSceneCurrentState();
+
+        public static void RegisterService(RpcServerPort<RPCContext> port)
+        {
+            if (!port.portName.StartsWith(REQUIRED_PORT_ID_START)) return;
+
+            RpcSceneControllerServiceCodeGen.RegisterService(port, new SceneControllerServiceImpl(port));
+        }
+
+        public SceneControllerServiceImpl(RpcServerPort<RPCContext> port)
+        {
+            port.OnClose += OnPortClose;
+            this.port = port;
+
+            sendCrdtMemoryStream = new MemoryStream();
+            sendCrdtBinaryWriter = new BinaryWriter(sendCrdtMemoryStream);
+
+            getStateMemoryStream = new MemoryStream();
+            getStateBinaryWriter = new BinaryWriter(getStateMemoryStream);
+        }
+
+        private void OnPortClose()
+        {
+            port.OnClose -= OnPortClose;
+
+            if (context != null && context.crdt.WorldState.ContainsScene(sceneNumber))
+                UnloadScene(null, context, new CancellationToken());
```

Comment:
Add `Forget()` if you are not going to use the results of `UniTask`

Reply:
tackled [here](https://github.com/decentraland/unity-renderer/pull/3700/commits/2a54ca9483686130b9aad39263bd6ac3209b579d)